File: tests/conformance/sandbox-tools/fetch_no_network.js
export default {
  description: "Tool with network disabled",
  input: {},
  permissions: {
    network: false,
  },
  execute: async (args, ctx) => {
    try {
      await ctx.fetch('http://localhost:18923/test');
      return { blocked: false };
    } catch (err) {
      return { blocked: true };
    }
  },
};

File: tests/conformance/sandbox-tools/fetch_no_network.py
tool = {
    "description": "Tool with network disabled",
    "input": {},
    "permissions": {
        "network": False,
    },
}


async def execute(args, ctx):
    try:
        await ctx.fetch("http://localhost:18923/test")
        return {"blocked": False}
    except Exception:
        return {"blocked": True}
